File: src/204CountPrimes/204CountPrimes.cpp
#include <iostream>
using namespace std;

class Solution {
public:
  int countPrimes(int n) {
    vector<bool> prime(n, false);
    int result = 0;
    for (int i = 2; i < n; i++) {
      if (!prime[i]) {
        result++;
        for (int j = 2; i * j < n; j++) {
          prime[i * j] = true;
        }
      }
    }
    return result;
  }
};

int main() {}

File: src/204CountPrimes/204CountPrimes.rs
impl Solution {
    pub fn count_primes(n: i32) -> i32 {
        let mut is_prime = vec![false; n as usize];
        let mut count = 0;
        for i in 2..n {
            if !is_prime[i as usize] {
                count += 1;
                for j in (i..n).step_by(i as usize){
                    is_prime[j as usize] = true;
                }
            }
        }
        count
    }
}
